plugins/module_utils/module_container/docker_api.py

```diff
@@ -269,8 +269,8 @@ def create_container(self, client, container_name, create_parameters):
     def start_container(self, client, container_id):
         client.post_json('/containers/{0}/start', container_id)
 
-    def wait_for_container(self, client, container_id):
-        return client.post_json_to_json('/containers/{0}/wait', container_id)['StatusCode']
+    def wait_for_container(self, client, container_id, timeout=None):
+        return client.post_json_to_json('/containers/{0}/wait', container_id, timeout=timeout)['StatusCode']
```

Comment:
passing `timeout=None` here basically disables timeout.

Reply:
(This is also what Docker SDK for Python does: https://github.com/docker/docker-py/blob/main/docker/api/container.py#L1288-L1320)